website/content/en/highlights/2021-11-04-remap-dropped-output.md

```diff
@@ -0,0 +1,82 @@
+---
+date: "2021-11-04"
+title: "New `dropped` output on the `remap` transform"
+description: "Sending failed events down a separate pipeline"
+authors: ["lukesteensen"]
+pr_numbers: [9169]
+release: "0.18.0"
+hide_on_release_notes: false
+badges:
+  type: "new feature"
+  domains: ["remap"]
+---
+
+Vector 0.18 introduces a new `dropped` output to the `remap` transform. This
+can be used to route events that fail processing down a separate pipeline.
+
+When either of `drop_on_error` or `drop_on_abort` is set to `true`, events that
+are dropped from the primary output stream due to either errors or aborts are
+written instead to a separate output called `dropped`.
+
+## Example
+
+As an example, the `dropped` output can be used if you want to capture events
+that failed during processing and send them out via a separate sink from the
+data the was processed successfully.
+
+Given a config of:
+
+```toml
+[sources.in]
+  type = "generator"
+  format = "shuffle"
+  interval = 1.0
+  lines = [
+    '{ "message": "valid message", "foo": "bar"}',
+    '{ "message": "valid message", "foo": "baz"}',
+    'invalid message',
+  ]
+
+[transforms.remap]
+  type = "remap"
+  inputs = ["in"]
+  drop_on_error = true
+  drop_on_abort = true
+  source = """
+    . |= object!(parse_json!(.message))
+    if .foo == "baz" {
+        abort
+    }
+    .processed = true
+  """
+
+[sinks.out]
+  type = "console"
+  inputs = ["remap"]
+  encoding.codec = "json"
+
+[sinks.errors_out]
+  type = "console"
+  inputs = ["remap.dropped"]
+  encoding.codec = "json"
+```
+
+You would expect to see output like the following:
+
+```json
+{"foo":"bar","message":"valid message","processed":true,"timestamp":"2021-11-05T00:42:03.945157398Z"}
+{"foo":"bar","message":"valid message","processed":true,"timestamp":"2021-11-05T00:42:04.945155276Z"}
+{"message":"{ \"message\": \"valid message\", \"foo\": \"baz\"}","metadata":{"component":"remap","error":"aborted"},"timestamp":"2021-11-05T00:42:05.945588208Z"}
```

Comment:
I think users might be used to seeing the same component tags that we have in other places (like metrics). Could we add the other tags here (`component_id`, `component_kind`, and `component_type`?). This would make:

```json
{"message":"{ \"message\": \"valid message\", \"foo\": \"baz\"}","metadata":{"component":"remap","error":"aborted"},"timestamp":"2021-11-05T00:42:05.945588208Z"}
```

Look like:

```json
{"message":"{ \"message\": \"valid message\", \"foo\": \"baz\"}","metadata":{"component_id":"remap","component_kind":"transform","component_type":"remap","error":"aborted"},"timestamp":"2021-11-05T00:42:05.945588208Z"}
```

I think this parity with metrics would be nice.